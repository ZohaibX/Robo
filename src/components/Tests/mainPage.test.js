import React from "react";
import Main from "../Main Page/main";
import renderer from "react-test-renderer"; // for snapshot

let tree;
beforeEach(() => {
  const mockProps = {
    searchField: "",
    robots: [],
    onSearchChange: jest.fn(),
  };
  tree = renderer.create(<Main {...mockProps} />).toJSON();
});

// we provide the mock data , the component needs
it("renders main page without crashing", () => {
  expect(tree).toMatchSnapshot();
});
