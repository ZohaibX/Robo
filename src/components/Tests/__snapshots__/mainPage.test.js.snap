// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders main page without crashing 1`] = `
<div>
  <h1
    id="h1"
  >
    RoboFriends
  </h1>
  <div
    className="pa2"
  >
    <input
      className="pa3 ba b--green bg-lightest-blue"
      id=""
      name="search"
      onChange={[Function]}
      placeholder="Search Your Friend"
      type="search"
    />
  </div>
  <div
    style={
      Object {
        "border": "1px solid black",
        "height": "500px",
        "overflowY": "scroll",
      }
    }
  >
    <div
      className=""
    />
  </div>
</div>
`;
